From 73358493bb886186023eba70809449b27e762788 Mon Sep 17 00:00:00 2001
From: Tharik Anwar <tharanwa@qti.qualcomm.com>
Date: Wed, 12 Aug 2026 18:37:32 +0530
Subject: [PATCH 10/10] arm64: dts: qcom: Add IMX577 Camera DT changes for
 Talos Lyra EVK

Add IMX577 Camera DT changes for Talos Lyra EVK.

Upstream-Status: Pending

Signed-off-by: Tharik Anwar <tharanwa@qti.qualcomm.com>
---
 .../qcom/talos-camera-sensor-lyra-evk.dtsi    | 108 ++++++++++++++++++
 1 file changed, 108 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/talos-camera-sensor-lyra-evk.dtsi b/arch/arm64/boot/dts/qcom/talos-camera-sensor-lyra-evk.dtsi
index 96af77f4f0f3..977ef743e443 100644
--- a/arch/arm64/boot/dts/qcom/talos-camera-sensor-lyra-evk.dtsi
+++ b/arch/arm64/boot/dts/qcom/talos-camera-sensor-lyra-evk.dtsi
@@ -545,6 +545,114 @@ tl_slot9: qcom,cam-sensor9 {
 		clock-rates = <19200000>;
 		status = "ok";
 	};
+
+	/*CAM6: IMX577*/
+	tl_slot10: qcom,cam-sensor10 {
+		cell-index = <10>;
+		compatible = "qcom,cam-sensor";
+		csiphy-sd-index = <0>;
+		sensor-position-roll = <0>;
+		sensor-position-pitch = <0>;
+		sensor-position-yaw = <180>;
+		cam_vio-supply = <&vreg_s4a>;
+		regulator-names = "cam_vio";
+		power-domains = <&camcc TITAN_TOP_GDSC>;
+		rgltr-cntrl-support;
+		pwm-switch;
+		rgltr-min-voltage = <1800000>;
+		rgltr-max-voltage = <1800000>;
+		rgltr-load-current = <120000>;
+		gpio-no-mux = <0>;
+		pinctrl-0 = <&cam_sensor_mclk0_active
+		             &cam_sensor_active_rst0>;
+		pinctrl-1 = <&cam_sensor_mclk0_suspend
+		             &cam_sensor_suspend_rst0>;
+		gpios = <&tlmm 28 0>,
+			<&tlmm 40 0>;
+		gpio-reset = <1>;
+		gpio-req-tbl-num = <0 1>;
+		gpio-req-tbl-flags = <1 0>;
+		gpio-req-tbl-label = "CAM_MCLK10",
+		                     "CAMIF_RESET10";
+		cci-master = <0>;
+		clocks = <&camcc CAM_CC_MCLK0_CLK>;
+		clock-names = "cam_clk";
+		clock-cntl-level = "nominal";
+		clock-rates = <24000000>;
+		status = "okay";
+	};
+
+	/*CAM7: IMX577*/
+	tl_slot11: qcom,cam-sensor11 {
+		cell-index = <11>;
+		compatible = "qcom,cam-sensor";
+		csiphy-sd-index = <1>;
+		sensor-position-roll = <0>;
+		sensor-position-pitch = <0>;
+		sensor-position-yaw = <180>;
+		cam_vio-supply = <&vreg_s4a>;
+		regulator-names = "cam_vio";
+		power-domains = <&camcc TITAN_TOP_GDSC>;
+		rgltr-cntrl-support;
+		pwm-switch;
+		rgltr-min-voltage = <1800000>;
+		rgltr-max-voltage = <1800000>;
+		rgltr-load-current = <120000>;
+		gpio-no-mux = <0>;
+		pinctrl-0 = <&cam_sensor_mclk1_active
+		             &cam_sensor_active_rst1>;
+		pinctrl-1 = <&cam_sensor_mclk1_suspend
+		             &cam_sensor_suspend_rst1>;
+		gpios = <&tlmm 29 0>,
+			<&tlmm 54 0>;
+		gpio-reset = <1>;
+		gpio-req-tbl-num = <0 1>;
+		gpio-req-tbl-flags = <1 0>;
+		gpio-req-tbl-label = "CAM_MCLK11",
+		                     "CAMIF_RESET11";
+		cci-master = <1>;
+		clocks = <&camcc CAM_CC_MCLK1_CLK>;
+		clock-names = "cam_clk";
+		clock-cntl-level = "nominal";
+		clock-rates = <24000000>;
+		status = "okay";
+	};
+
+	/*CAM8: IMX577*/
+	tl_slot12: qcom,cam-sensor12 {
+		cell-index = <12>;
+		compatible = "qcom,cam-sensor";
+		csiphy-sd-index = <2>;
+		sensor-position-roll = <0>;
+		sensor-position-pitch = <0>;
+		sensor-position-yaw = <180>;
+		cam_vio-supply = <&vreg_s4a>;
+		regulator-names = "cam_vio";
+		power-domains = <&camcc TITAN_TOP_GDSC>;
+		rgltr-cntrl-support;
+		pwm-switch;
+		rgltr-min-voltage = <1800000>;
+		rgltr-max-voltage = <1800000>;
+		rgltr-load-current = <120000>;
+		gpio-no-mux = <0>;
+		pinctrl-0 = <&cam_sensor_mclk2_active
+		             &cam_sensor_active_rst2>;
+		pinctrl-1 = <&cam_sensor_mclk2_suspend
+		             &cam_sensor_suspend_rst2>;
+		gpios = <&tlmm 30 0>,
+			<&tlmm 42 0>;
+		gpio-reset = <1>;
+		gpio-req-tbl-num = <0 1>;
+		gpio-req-tbl-flags = <1 0>;
+		gpio-req-tbl-label = "CAM_MCLK12",
+		                     "CAMIF_RESET12";
+		cci-master = <1>;
+		clocks = <&camcc CAM_CC_MCLK2_CLK>;
+		clock-names = "cam_clk";
+		clock-cntl-level = "nominal";
+		clock-rates = <24000000>;
+		status = "okay";
+	};
 };
 
 &soc {
-- 
2.34.1

